packages/ui/src/TransactionSummary/use-classification.ts

```diff
@@ -0,0 +1,169 @@
+import { useMemo } from 'react';
+import { TransactionClassification } from '@penumbra-zone/perspective/transaction/classification';
+import { Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import { AddressView } from '@penumbra-zone/protobuf/penumbra/core/keys/v1/keys_pb';
+import { TransactionInfo } from '@penumbra-zone/protobuf/penumbra/view/v1/view_pb';
+import { classifyTransaction } from '@penumbra-zone/perspective/transaction/classify';
+import { findRelevantAssets } from '@penumbra-zone/perspective/action-view/relevant-assets';
+import {
+  IbcRelay,
+  Ics20Withdrawal,
+} from '@penumbra-zone/protobuf/penumbra/core/component/ibc/v1/ibc_pb';
+import { unpackIbcRelay } from '@penumbra-zone/perspective/action-view/ibc';
+import { GetMetadataByAssetId } from '../ActionView/types';
+import { isMetadata } from '../AssetSelector';
+
+interface SummaryData {
+  type: TransactionClassification;
+  label: string;
+  memo?: string;
+  additionalText?: string;
+  assets: Metadata[];
+  tickers?: string[];
+  address?: AddressView;
+}
+
+const DEFAULT_MEMO = 'Memo empty';
+
+const CLASSIFICATION_LABEL_MAP: Record<TransactionClassification, string> = {
+  unknown: 'Unknown',
+  unknownInternal: 'Unknown (Internal)',
+  receive: 'Receive',
+  send: 'Send',
+  internalTransfer: 'Internal Transfer',
+  ics20Withdrawal: 'Withdrawal',
+  ibcRelayAction: 'Deposit',
+  swap: 'Swap',
+  swapClaim: 'Swap Claim',
+  delegate: 'Delegate',
+  undelegate: 'Undelegate',
+  undelegateClaim: 'Undelegate Claim',
+  dutchAuctionSchedule: 'Auction Schedule',
+  dutchAuctionEnd: 'Auction End',
+  dutchAuctionWithdraw: 'Auction Withdraw',
+  delegatorVote: 'Vote',
+  validatorVote: 'Vote',
+  communityPoolDeposit: 'Community Pool Deposit',
+  communityPoolOutput: 'Community Pool Output',
+  communityPoolSpend: 'Community Pool Spend',
+  positionClose: 'Close Position',
+  positionOpen: 'Open Position',
+  positionWithdraw: 'Withdraw Position',
+  positionRewardClaim: 'Claim Position Reward',
+  proposalDepositClaim: 'Proposal Deposit Claim',
+  proposalSubmit: 'Create Proposal',
+  proposalWithdraw: 'Withdraw Proposal',
+  validatorDefinition: 'Validator Definition',
+  liquidityTournamentVote: 'Liquidity Tournament Vote',
+};
+
+/**
+ * A hook that prepares data from TransactionInfo to be rendered in TransactionSummary.
+ */
+export const useClassification = (
+  info: TransactionInfo,
+  getMetadataByAssetId?: GetMetadataByAssetId,
+) => {
```

Comment:
the main data modeling logic is here - the hook prepares a transaction for rendering